lib/streamlit/credentials.py

```diff
@@ -163,6 +163,10 @@ def reset(cls):
 
     def save(self):
         """Save to toml file."""
+        # Create intermediate directories if necessary
+        os.makedirs(os.path.dirname(self._conf_file), exist_ok=True)
```

Comment:
Do we need to store references to multiple `_conf_file`s instead of only to the one at user home? Are things like `reset` affected by the conf file in the project directory?

Reply:
Credentials are stored in a separate file (`~/.streamlit/credentials.toml`), and there's no support for "project-local" credentials. (This bit of logic is unrelated to the config changes, but it was necessary for the `util. get_streamlit_file_path` changes.)